src/vs/workbench/services/editor/common/editorGroupsService.ts

```diff
@@ -214,6 +213,11 @@ export interface IEditorGroupsContainer {
 	 */
 	readonly onDidChangeGroupLocked: Event<IEditorGroup>;
 
+	/**
+	 * An event for when the locked state of a group changes.
+	 */
+	readonly onDidMaximizeGroup: Event<{ group: IEditorGroup; maximized: boolean }>;
```

Comment:
Since the `maximized` state is part of the event, I would rename this to `onDidChangeGroupMaximized`